airbyte-api/src/main/openapi/config.yaml

```diff
@@ -2789,6 +2789,8 @@ components:
           format: uuid
         disable_cache:
           type: boolean
+        notifySchemaChange:
```

Comment:
This doesn't feel right, I don't think we want to expose this flag here, it keeps on adding to the fact that our APIs are sometimes doing too many things and we should try to rationalize into simpler operations.
It might be a good time to extract the notification code into its own class to avoid this, what do you think?

Reply:
What do you mean exactly? this is here because we do want to notify for automatic schema changes but not for manual schema changes, so it's an option tied to the source schema refresh

Reply:
Do we want to keep going through the API for the automatic refresh? I thought this should only be used for manual schema change.
We currently have handlers depending on handlers which is a dependencies we should remove, it looks like we're exposing more internals here due to this bad dependency. I am wondering if we can have a solution that doesn't involve exposing more config not intended to be exposed in the first place.

Reply:
I think at this point, yes, partly since we want to get this out this week. Maybe we should jump on a call about this since I'm not sure exactly what kind of refactor you're thinking of - but is it ok if we refactor afterward?

Reply:
@gosusnp I think that the main issue here is that the API is doing a lot here. Another fix would be to also have another endpoint that would be internal (maybe this one and another one in the webbackend for the manual operation). Should we exclude that from the scope of the ticket?